src/gui/input/InputAPIAddWindow.h

```diff
@@ -50,4 +51,10 @@ class InputAPIAddWindow : public wxDialog
 
 	std::vector<ControllerPtr> m_controllers;
 	std::atomic_bool m_search_running = false;
+	struct AsyncThreadData
+	{
+		std::atomic_bool discardResult = false;
+		std::mutex mutex;
```

Comment:
Suggestion: Make the mutex a member variable of `InputAPIAddWindow`. Tracking it per-thread only unnecessarily complicates things and is error prone.

Reply:
Actually, this may then introduce a race condition where `this` is destroyed before the thread accesses it. Not sure, maybe there is a less fragile way to handle this in general.

Reply:
> maybe there is a less fragile way to handle this in general.

Have a look at the latest commit. I switched to a jthread. I think I was reinventing the wheel.

Reply:
I've reverted it. Not only because of the lack of macOS support but also because it has more theoretical race conditions than the mutex.